# Twitter Network Analysis of Competing Cosmetic Brands
#### APRD6343-0001: Advanced Statistics <br> Michelle Puglio<br>April 2019
The goal of this project is to better understand the virtual "chatter" around each brand, what makes each brand unique, and what makes each brand different. Network Anaysis will also be used to identify users that are most central to the brand(s) and ones that are hyper interested in the product category (__cosmetics__) in general.

### Selected Brands:
> [Too Faced](https://www.toofaced.com/)<br>
> [Urban Decay](https://www.urbandecay.com/)<br>
> [MAC Cosmetics](https://www.maccosmetics.com/)

__Part One: Fetching Twitter Data.__ <br> Using the Twitter API, specifically the "Search Endpoint", retrieve Tweets that mention the three chosen brands (*Too Faced, Urban Decay, MAC Cosmetics*). The API returns historical results for the last 7 days. That is, this code will return all of the tweets sent by consumers (and other twitter users) that mention these brands' names. Fetch all available for each brand and save those tweets as JSON inside a zip file. Three zipfiles should be stored in the folder, one for each brand.

Code: [twittersearch.py](https://aprd6343.codeslick.com/user/mipu8362/edit/Network%20Analysis/1Final%20Report/twittersearch.py)

__Part Two: Twitter Mentions Graph.__ <br> Using python - create a valued, directed network graph of twitter mentions. The graph should show Twitter users that are most centrally related to the brand (e.g., they regularly mention the brand). This graph should also illustrate who mentions who on Twitter, and in what way those mentions flow. Create one mention graph, and that mention graph will have mentions for all three brands.

For comprehensive purposes, I chose to only include users who had over 1,000 followers within the Twitter Mentions edgelist. The decision behind this reasoning can be broken down into two parts:
> __1)__ Accounts like these are more socially/technologically "relevant" and therefore, influential. With at least a thousand people (potential consumers) or more viewing the content produced via these accounts daily, they are probably the most revelant to the companies whose products they are tweeting about - (since the opinions on these accounts are reaching the most people).<br>
> __2)__ In terms of modeling, the Gephi graphs became clearer and clearer as I was able to reduce the number of nodes. (Less *was* more.) Once the edgelist was segmented to only include the most influential accounts - I was able to produce the most readable, comprehensible, and intuitive graphs.

__2.1) Build mention network edgelist.__

In [160]:
import glob
import os
import shutil
import zipfile
import json
import csv

In [161]:
TMPDIR = 'tmp'

In [162]:
if not os.path.exists(TMPDIR):
    os.makedirs(TMPDIR)

In [163]:
tweetzipfiles = glob.glob('*.zip')
tweetzipfiles

['too_faced_OR_toofaced.zip',
 'urban_decay_OR_urbandecay.zip',
 'mac_cosmetics_OR_maccosmetics.zip']

In [164]:
for tweetzipfile in tweetzipfiles:
    with zipfile.ZipFile(tweetzipfile, "r") as f:
        print('Unzipping to tmp directory: %s' % tweetzipfile)
        f.extractall(TMPDIR)

Unzipping to tmp directory: too_faced_OR_toofaced.zip
Unzipping to tmp directory: urban_decay_OR_urbandecay.zip
Unzipping to tmp directory: mac_cosmetics_OR_maccosmetics.zip


In [165]:
uniqueusers = {}
count = 0
for fn in os.listdir(TMPDIR):
    fn = os.path.join(TMPDIR, fn)
    with open(fn) as f:
        count += 1
        if count % 1000 == 0:
            print(count)
        tweetjson = json.load(f)
        userwhotweeted = tweetjson['user']['screen_name']
        followercount = tweetjson['user']['followers_count']
# only include users with more than 1,000 followers
        if followercount > 1000:
            print(followercount)
            if userwhotweeted in uniqueusers:
                uniqueusers[userwhotweeted] += 1
            if userwhotweeted not in uniqueusers:
                uniqueusers[userwhotweeted] = 1

2747
2344
1548069
3877
1150
1488
3175
1519
3549
21342
9206
1816
1232
1731
3945
1164
71938
1204
2018
3099
1538
9547
4070
36298
1181
1378
1122
1525
3166
1207
1178
1639
2718
4424
1522
3358
49653
1368
1609
5533
9335
1207
1001
4744
1819
14981
2035
3171
2812
2970
4458
2542
5811
2083
5812
1563
2600
7619
1856
1046
2266
2804
10208
11691
19980
7650
2742
35495
1303
2086
1358
1194
10884
1128
6115
3054
47676
3285
4245
8790
4686
1049
1192
4419498
2188
2106
12861
16665
3081
4655
1480
2151
8567
1187
1214
2120
5068
6581
2569
1582
3403
1368
1360
1003
1206
66814
1996
1709
2119
7339
1182
5468
1903
3768
1274137
2106
8109
7556
1259
1719
1893
3354
1130
2262
1183
1674
1039049
6035
7235
2261
1465
1841
18168
5068
10797
1125
4503
2887873
1581
3660
2949
5590
2019
21144
1153
5046
1104
5371
2106
3681
1438
1411
5433
1275
1243
2742
5460
2263
2447
4683
215838
5068
1259
2742
1571
13676
1044
7031
7556
4220
3205
3104
1368
2497
9515
2082
1402
1678
14083
22201
5139
2380
35772
1135
15308
5013
19818
2985
1269
1034
3047
6673


2306
2991
1084
5433
3962
1168
1004
3769
1750
6081
1164
2463
2402
3974
2826
3665
1197
1672
1303
2261
3167
2354
1240
2985
1878
2098
1198
1369
1039049
1315
2260
3146
1780
5614
1307
4073
3351
5068
4314
17896
1124
1234
2643
1438
2769
26711
16585
1688
1908
1821
10705
1915
3524
2322
1808
2167
1025
5533
2768
1550
3337
286696
3769
2751
2319
3930
1425
1584
3243
11703
2263
18248
3952
7016
33252
2720
36638
1187
5923
290786
2303
1812
50667
3150
996784
1378
5923
18707
11690
2129
1548069
1031
1207
1734
2028
1397
6350
3167
2791
1026
1153
4411
2431
1037
4912
1969
7257
6000
1088
1646
1735
8409
2263
2018
3443
7265
1527
36160
1106
9426
4576
2742
36160
1485
1218
5295
7650
2807
29592
2261
26558
1034
16913
1102
10542
3961
5911
4266
2938
18129
1686
2064
1735
5758
2483
5433
32915
2669
1028
5433
12074
12719
41196
3271
1706
363869
1859
336896
3165
7016
1365
4405
61280
1636
45031
1150
1798
2658
1950
11649
212383
1173
3038
10409
4147
1730
171535
3331
13542
1226
2263
3430
1287
5068
12983
1045
2518
1911
1577
111476


5015
1786
2048
1762
6859
1507
3167
2098
1218
596862
1715
13873
146044
12203
3769
1989
1478
14217
1475
3743
1013
69100
3159
40034
3775
1799
15824
12118
20578
1261
3769
1994
1259
1311
239029
1162
2256
33212
1126
1121
13720
1261
7376
1549
1316134
13957
26908
17945
1241
1262
10738
3743
8335
23667
195632
2770
3202
1415
1564
13038
7154
3558
4720
1219
1341
1049
2018
2694
3241
3891
1422
7847
78135
205191
1670
4519
4419498
4930
3668
12123
3769
2262
1277
200586
4527
3059
3221
1026
5229
1290
1917
1363
19675
1925
22164
3524
1637
996784
1546
9293
10688
1161
1031
11690
1071
1907
3205
24141
3873
1193
2876
4419497
1259
2269
7474
1855
1200
1171
1542
3769
5299
2711
9514
1233
1166
5789
21829
2363
6936
6091
1850
13843
31675
3493
37784


In [166]:
len(uniqueusers)

2892

In [167]:
userstoinclude = set()
usercount = 0
for auser in uniqueusers:
    if uniqueusers[auser] > 1:   # parameter
        usercount += 1
        userstoinclude.add(auser)
        
        
print(len(userstoinclude))

264


In [168]:
edgelist = open('makeupbrands.edgelist.for.gephi.csv', 'w')
csvwriter = csv.writer(edgelist)
header = ['Source', 'Target']
csvwriter.writerow(header)

15

In [169]:
print('Writing edge list')
count = 0
for fn in os.listdir(TMPDIR):
    fn = os.path.join(TMPDIR, fn)
    with open(fn) as f:
        tweetjson = json.load(f)
        userwhotweeted = tweetjson['user']['screen_name']
        if userwhotweeted in userstoinclude:
            count += 1
            if count % 1000 == 0:
                print(count)
            
            users = tweetjson['entities']['user_mentions']
            if len(users) > 0:
                for auser in users:
                    screenname = auser['screen_name']
                    row = [userwhotweeted, screenname]
                    csvwriter.writerow(row)
edgelist.close()

Writing edge list


__2.2) Create twitter mentions graph.__

In [170]:
# Graphs with Gephi

__Twitter Mentions Graph (Overview)__
<img src="Twitter Mention Graph - Overview.png">

### Who are the Most Important Bridgers?
Some users will mention multiple (or all three) brands. Who are those people? If bridge users (bridgers) exist, who are they? Why are they interested in this product so much? From an advertising/marketing perspective, these are people who are "pro users" of your category, and their hyper interest is something that you can use to engage with them, that is, they'll always be interested in your new product/services before others are.

In the Yifan Hu Proportional (Gephi) Twitter Network Analysis model presented above, we identify three major bridgers:
> [@InStyle](https://twitter.com/InStyle?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor)<br>
*connects Urban Decay and Too Faced*<br>
Followers = 4.42M<br>
Interest: American Beauty Tips, Celebrity Style, and Fashion Advice Magazine.

> [@beautytrends24](https://twitter.com/beautytrends24)<br>
*connects Too Faced and MAC Cosmetics*<br>
Followers = 2,709<br>
Interest: American Twitter beauty influencer, recieves attention/rewards from beauty brands for posting content about their products.

> [@Yruil_sg](https://twitter.com/Yuril_sg)<br>
*connects Urban Decay and MAC Cosmetics*<br>
Followers = 1,343<br>
Interest: Singaporean Twitter beauty influencer, recieves attention/rewards from beauty brands for posting content acout their products.


### Who are the Most Central Users?
In looking at the directed, valued network of mentions, what users are most central for each brand? Inspect at least the top 10 most central for each brand. Who are they? Why do you think they're so engaged with your brand on Twitter?

From an advertising/marketing perspective, these are our loyal brand ambassadors. A good social media manager should know who these people are and give them preferential treatment, such as sending them swag, sending them early access to new products/content, and generally engaging with them on the platform to show reciprocity in the relationship.

__Twitter Mentions Graph (Too Faced)__
<img src="Too Faced Mention Graph.png">

*Most Central Users*:<br>
> [@InStyle](https://twitter.com/InStyle?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor)<br>
Followers = 4.42M<br>
Interest: American Beauty Tips, Celebrity Style, and Fashion Advice Magazine.

> [@vogueaustrailia](https://twitter.com/vogueaustralia)<br>
Followers = 176K<br>
Interest: Austrailian magazine for fashion, style, and beauty.

> [@Laura_Nolan22](https://twitter.com/search?q=Laura_Nolan22&src=typd)<br>
Followers = 23.1K<br>
Interest: English Beauty, Fashion & Lifestyle Blogger & YouTuber.

> [@MyBeautyBunny](https://twitter.com/search?q=MyBeautyBunny&src=typd)<br>
Followers = 50.7K<br>
Interest: An award-winning, top 10, cruelty-free beauty, travel, health, lifestyle and fashion blog. My Beauty Bunny focuses on finding the best in beauty from companies who do not test on animals.

> [@sarah_berryman1](https://twitter.com/sarah_berryman1)<br>
Followers = 11.7K<br>
Interest: English Beauty Blogger & Mum of 2. Finalist in The New Generation Beauty Influencer Awards 2017. 

> [@acurvyvegan](https://twitter.com/acurvyvegan)<br>
Followers = 10.6K<br>
Interest: Award-winning blogger and content creator from the UK.

> [@ClosetKate](https://twitter.com/ClosetKate)<br>
Followers = 6,382<br>
Interest: Blogger/Lover of clothes, skincare, and makeup from the UK.

> [@sotonbloggers](https://twitter.com/sotonbloggers)<br>
Followers = 4,467<br>
Interest: Beauty blogging group in Southampton.

> [@BBlogRT](https://twitter.com/BBlogRT)<br>
Followers = 36.1K<br>
Interest: A promotional network and online community for bloggers to promote their work.

> [@bloggingbeesrt](https://twitter.com/bloggingbeesrt)<br>
Followers = 3,107<br>
Interest: Twitter chat and #RT account. Connecting bloggers all over.

__Twitter Mentions Graph (Urban Decay)__
<img src="Urban Decay Mention Graph.png">

*Most Central Users*:<br>
> [@InStyle](https://twitter.com/InStyle?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor)<br>
Followers = 4.42M<br>
Interest: American Beauty Tips, Celebrity Style, and Fashion Advice Magazine.

> [@Yruil_sg](https://twitter.com/Yuril_sg)<br>
Followers = 1,343<br>
Interest: Singaporean Twitter beauty influencer, recieves attention/rewards from beauty brands for posting content acout their products.

> [@Makeupdotcom](https://twitter.com/Makeupdotcom)<br>
Followers = 40.7K<br>
Interest: New York City based makeup blog, sponsored by L'Oreal.

> [@lil_yenta](https://twitter.com/lil_yenta)<br>
Followers = 10.7K<br>
Interest: Philidelphia blogger/influencer.

> [@Nicia_K](https://twitter.com/Nicia_K)<br>
Followers = 1,735<br>
Interest: Youtuber and beauty-blogger based in Texas.

> [@ohsixeleven](https://twitter.com/ohsixeleven)<br>
Followers = 2,034<br>
Interest: UK blogger writing about travel, beauty & lifestyle.

> [@LovingBlogs](https://twitter.com/LovingBlogs)<br>
Followers = 27.7K<br>
Interest: A promotional network and for bloggers to promote their work.

__Twitter Mentions Graph (MAC Cosmetics)__
<img src="MAC Mention Graph.png">

*Most Central Users*:<br>
> [@Revelere](https://twitter.com/Revelere)<br>
Followers = 3,758<br>
Interest: Curating the best pricing on luxe beauty brands you love in skincare, hairstyle, makeup, nail polish, and more from around the globe.

> [@jetsetlaura](https://twitter.com/jetsetlaura)<br>
Followers = 225<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@laurapasco](https://twitter.com/laurapasco)<br>
Followers = 76<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@BrunetteJenner](https://twitter.com/BrunetteJenner)<br>
Followers = 99<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@kaitbxwen](https://twitter.com/kaitbxwen)<br>
Followers = 1,521<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@martina_ox](https://twitter.com/martina_ox)<br>
Followers = 7<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@ShanHarrison_x](https://twitter.com/ShanHarrison_x)<br>
Followers = 281<br>
Interest: Consumer, has interacted with the official MAC Cosmetics account directly.

> [@XingPark](https://twitter.com/XingPark)<br>
Followers = 111K<br>
Interest: Famous Chinese singer/song-writer and MAC's first ever spokesperson for China.

__Part Three: Semantic Network Graph.__ <br> Next, create a semantic network analysis graph of words used in Tweets. Practically, this graph should reveal what words are most commonly associated with each other, for each brand. Create one semantic graph, and that graph will have mentions for all three brands.

__3.1) Build semantic network edgelist.__

In [171]:
# nuke TMPDIR
shutil.rmtree(TMPDIR)

In [172]:
import nltk
wn = nltk.WordNetLemmatizer()
ps = nltk.PorterStemmer()
import glob
import os
import re
import shutil
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
nltk.download('stopwords')
nltk.download('punkt')
nltk.download('wordnet')
import csv
from nltk.stem import WordNetLemmatizer
wordnet_lemmatizer = WordNetLemmatizer()
import string
import itertools
import zipfile
import json
punctuation = string.punctuation
stopwordsset = set(stopwords.words("english"))
stopwordsset.add("'s")

#Removing urls
def removeURL(text):
    result = re.sub(r"http\S+", "", text)
    return result

#Extracting contextual words from a sentence
def tokenize(text):
    #lower case
    text = text.lower()
    #split into individual words
    words = word_tokenize(text)
    return words

def stem(tokenizedtext):
    rootwords = []
    for aword in tokenizedtext:
        aword = ps.stem(aword)
        rootwords.append(aword)
    return rootwords

def stopWords(tokenizedtext):
    goodwords = []
    for aword in tokenizedtext:
        if aword not in stopwordsset:
            goodwords.append(aword)
    return goodwords

def lemmatizer(tokenizedtext):
    lemmawords = []
    for aword in tokenizedtext:
        aword = wn.lemmatize(aword)
        lemmawords.append(aword)
    return lemmawords

def removePunctuation(tokenizedtext):
    nopunctwords = []
    for aword in tokenizedtext:
        if aword not in punctuation:
            nopunctwords.append(aword)
    cleanedwords = []
    for aword in nopunctwords:
        aword = aword.translate(str.maketrans('', '', string.punctuation))
        cleanedwords.append(aword)
    return cleanedwords

[nltk_data] Downloading package stopwords to
[nltk_data]     /home/mipu8362/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to /home/mipu8362/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package wordnet to /home/mipu8362/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!


In [173]:
TMPDIR = 'tmp'

In [174]:
if not os.path.exists(TMPDIR):
    os.makedirs(TMPDIR)

In [175]:
tweetzipfiles = glob.glob('*.zip')

In [176]:
for tweetzipfile in tweetzipfiles:
    with zipfile.ZipFile(tweetzipfile, "r") as f:
        print("Unzipping to tmp directoru: %s" % tweetzipfile)
        f.extractall(TMPDIR)

Unzipping to tmp directoru: too_faced_OR_toofaced.zip
Unzipping to tmp directoru: urban_decay_OR_urbandecay.zip
Unzipping to tmp directoru: mac_cosmetics_OR_maccosmetics.zip


In [177]:
uniquewords = {}
count = 0
for fn in os.listdir(TMPDIR):
    fn = os.path.join(TMPDIR, fn)
    with open(fn) as f:
        tweetjson = json.load(f)
        count += 1
        if count % 1000 == 0:
            print(count)
            
        text = tweetjson['text']
        language = tweetjson['metadata']['iso_language_code'] # only include English words
        if language == "en":
            nourlstext = removeURL(text)
            tokenizedtext = tokenize(nourlstext)
            nostopwordstext = stopWords(tokenizedtext)
            lemmatizedtext = lemmatizer(nostopwordstext)
            nopunctuationtext = removePunctuation(lemmatizedtext)

            for aword in nopunctuationtext:
                if aword in uniquewords:
                    uniquewords[aword] += 1
                if aword not in uniquewords:
                    uniquewords[aword] = 1

1000
2000
3000
4000
5000
6000
7000
8000
9000
10000
11000


In [178]:
wordstoinclude = set()
wordcount = 0
for aword in uniquewords:
    if uniquewords[aword] > 25:
        wordcount += 1
        wordstoinclude.add(aword)

In [179]:
# shoot for ~ 1000 words by tweet the greater than parameter in the above cell
print(wordcount)

731


In [180]:
edgelist = open('makeupbrands.semantic.edgelist.for.gephi.csv', 'w')
csvwriter = csv.writer(edgelist)

header = ['Source', 'Target', 'Type']
csvwriter.writerow(header)

print('Writing edge list')
uniquewords = {}
count = 0
for fn in os.listdir(TMPDIR):
    fn = os.path.join(TMPDIR, fn)
    with open(fn) as f:
        tweetjson = json.load(f)
        count += 1
        if count % 1000 == 0:
            print(count)
            
        text = tweetjson['text']
        nourlstext = removeURL(text)
        tokenizedtext = tokenize(nourlstext)
        nostopwordstext = stopWords(tokenizedtext)
        lemmatizedtext = lemmatizer(nostopwordstext)
        nopunctuationtext = removePunctuation(lemmatizedtext)
        
        goodwords = []
        for aword in nopunctuationtext:
            if aword in wordstoinclude:
                goodwords.append(aword.replace(',',''))
                
        allcombos = itertools.combinations(goodwords, 2)
        for acombo in allcombos:
            row = []
            for anode in acombo:
                row.append(anode)
            row.append('Undirected')
            csvwriter.writerow(row)
edgelist.close()

Writing edge list
1000
2000
3000
4000
5000
6000
7000
8000
9000
10000
11000


In [181]:
# nuke directory
shutil.rmtree(TMPDIR)

__3.2) Create semantic network graph.__

In [182]:
# Graph with Gephi

__Semantic Network Graph (Overview)__
<img src="Semantic Network Graph - Overview.png">

## Look at Bridging Words
Brands will likely share some key attributes (words). If so, what are they? What commonalities do brands have with each other? These commonalities can be thought of as product differentiation things to avoid. That is, we'd never create advertising or marketing materials that reflect non-unique attributes about our products. Knowing that we share things with our competitors also can tell us how we better need to differentiate our products.

In the Yifan Hu Proportional (Gephi) Semantic Network model presented above, we identify two major bridging words (key attributes) that these brands share:
> "de"<br>
*Shared by all three brands.*<br>
__Assumed Key Attributes:__ Too Faced, Urban Decay, and MAC Brands all possess a marketing platform catered to Spanish-speaking consumers/countries.

> "rt"<br>
*Shared by all three brands.*<br>
__Assumed Key Attributes:__ Too Faced, Urban Decay, and MAC Brands additionally all possess
a strong "brand presence" on Twitter - implied by their strong assimilations with the word    "retweet" (rt) which means their content is commonly re-shared on Twitter.

Although the question above implies that commonalities between brands should be avoided - in this particular case - I do not recommend that any of these comestic brands (TooFaced, Urban Decay, or MAC) steer *any* marketing focus away from creating/maintaining a strong online Twitter presence or discontinue their market focus to non-English speaking (particularly, Spanish-speaking) countries and consumers.

## Inspect Semantic Word Clusters
Look at the word clusters. Interpret them. Identify the key clusters of activity. What's going on in each cluster?

## What Word Clusters Does Each Brand Own?
What attributes (words, or cluster of words) does each brand uniquely own? That is, what attributes are uniquely linked to each brand? From a marketing perspective, the most central nodes are the attributes that each brand most "owns." These are implicit to the consumer, and therefore our strengths. Practically, this means that the brand doesn't need to do work advertising/marketing these perceptions, instead it should focus on reinforcing things that aren't as central, or missing from this semantic network completely.

__Semantic Word Cluster (Too Faced)__
<img src="TooFaced Semantic Word Cluster.png">

*Too Faced "Owned" Words*:<br>
 - __one__
 - __trouble__
 - __better__
 - __sex__
 - __way__
 - __born__

These attributes that Too Faced "owns" and are uniquely linked to on Twitter imply that the brand's "strength(s)" are product-based. More specifically, Too Faced Cosmetics Brand is best well-known for their ["Born This Way"](https://www.toofaced.com/face-makeup/foundation/born-this-way-foundation-swan/70201.html#q=trouble&lang=default&start=1) foundation and the iconic ["Better Than Sex"](https://www.toofaced.com/eye-makeup/mascara/better-than-sex-mascara/80021.html?cgid=mascara#q=sex&lang=default&start=1) mascara.

__Semantic Word Cluster (Urban Decay)__
<img src="Urban Decay Semantic Word Cluster.png">

*Urban Decay "Owned" Words*:<br>
 - __follow__
 - __giveaway__
 - __nighter__
 - __30ml__
 - __naked__
 - __cherry__
 - __pallet__
 - __reloaded__

Similarly to Too Faced, these attributes that Urban Decay "owns" and are uniquely linked to on Twitter imply that the brand's "strength(s)" are generally product-based. More specifically, Urban Decay Cosmetics Brand is best well-known for their ["Naked Reloaded"](https://www.urbandecay.com/naked-reloaded-urban-decay/ud938.html#q=reloaded&start=3) and ["Naked Cherry"](https://www.urbandecay.com/naked-cherry-eyeshadow-palette-urban-decay/ud927.html) eye shadow pallets.

However, Urban Decay is also closely associated with the words "follow" and "giveaway" - which implies a higher level of interaction between the brand's Twitter account and its followers (particularly in regards to online giveaways and promotions).

__Semantic Word Cluster (MAC Cosmetics)__
<img src="MAC Semantic Word Cluster.png">

*MAC Cosmetics "Owned" Words*:<br>
 - __zhang__
 - __region__
 - __china__
 - __spokesperson__
 - __hello__
 - __everyone__

On the MAC Comestics [website](https://www.maccosmetics.com/our-story), the brand describes itself as "the world's leading professional makeup authority" and one that "celebrates diversity and __INDIVIDUALITY__" as their products "are for ALL ages, ALL Races, ALL Genders". The *Who We Are* page goes on to say: "MAC means so many things to its __millions of fans in every corner of the planet__, but a global beauty phenomenon has to begin somewhere". MAC Cosmetics is a cosmetics brand that seeks to cater to *global* beauty trends, fashions, and artistry.

Quite differently than Too Faced or Urban Decay, whose market strengths are product-based, the attributes that MAC Cosmetics Brand "owns" and are uniquely linked to on Twitter imply that their success derives from efforts to advertise to the "global" beauty market. Not one product fits all - rather, products for everyone (i.e., [Zhang Yixing for MAC Cosmetics China](https://exo-stentialism.tumblr.com/post/177712052701/zhang-yixing-for-mac-cosmetics-china)). This is why we see words like "region", "spokesperson", "china", "zhang", and "everyone" in MAC's semantic word cluster.